src/nspawn/nspawn.c

```diff
@@ -676,9 +676,8 @@ static int parse_argv(int argc, char *argv[]) {
                                 r = free_and_strdup(&arg_machine, optarg);
                                 if (r < 0)
                                         return log_oom();
-
-                                break;
                         }
+                        break;
```

Comment:
Well, dunno. That's what seems to have been the intent of the code. I don't care much either way, but even if we disallow `-M ''`, I think this can/should be a separate issue.